$NetBSD: patch-aa,v 1.3 2005/09/27 11:25:36 dsainty Exp $

Add support for the PIC16F628A.  Same as the PIC16F628, but the device ID and
mask have changed, and the screen template needed to be adjusted for the
longer device name.

--- picprg.c	2002-05-21 03:20:02.000000000 +1200
+++ picprg.c	2005-09-13 01:45:06.000000000 +1200
@@ -154,10 +154,24 @@
 NULL
 };
 
+char *pic16F628a_config_strings[] = {
+"CLOCK,0x0013,14,8,0x00:LP,0x01:XT,0x02:HS,0x03:EXTCLK,0x10:INTRCIO," //cont...
+"0x11:INTRCCLK,0x12:EXTRCIO,0x13:EXTRCCLK",
+"CP,0x3c00,30,4,0x0000:ALL,0x3c00:OFF,0x2800:HALF,0x1400:3/4",
+"WD,0x0004,39,3,0x0004:ON,0x0000:OFF",
+"PU,0x0008,47,3,0x0008:OFF,0x0000:ON",
+"CPD,0x0100,55,3,0x0100:OFF,0x0000:ON",
+"LVP,0x0080,64,3,0x0080:ON,0x0000:OFF",
+"BOD,0x0040,110,3,0x0040:ON,0x0000:OFF",
+"MCLR,0x0020,119,3,0x0020:EXT,0x0000:INT",
+NULL
+};
+
 dev_id_t pic_dev[] =
 {{0x07c0,0x0fc0,0x0800,0x0080,0x10,"PIC16F628",pic16F628_config_strings},
  {0x09a0,0x0fe0,0x2000,0x0100,0x10,"PIC16F877",pic16F877_config_strings},
  {0x0560,0x0fe0,0x0400,0x0040,0x00,"PIC16F84A",pic16F84_config_strings},
+ {0x1060,0x3fe0,0x0800,0x0080,0x10,"PIC16F628A",pic16F628a_config_strings},
  {0x0000,0x0000,0x0000,0x0000,0x00,NULL}
 };
 
